src/jasonwynn10/Nicknames/Main.php

```diff
@@ -58,11 +62,22 @@ public function onCommand(CommandSender $sender, Command $cmd, string $label, ar
 			$this->nicknameDB->set($target, $target);
 			$this->nicknameDB->save();
 		}else {
-			$sender->setDisplayName($target);
+		    if (!$this->getConfig()->get("economy") or $this->getServer()->getPluginManager()->getPlugin("EconomyAPI")) {
+		        $sender->setDisplayName($target);
 			$sender->sendMessage(TextFormat::GREEN."Nickname set to ".$args[0]);
 			$this->nicknameDB->set($target, $args[0]);
 			$this->nicknameDB->save();
+		    }
+		    if (!$this->economy->myMoney($player) >= $this->getConfig()->get("buyprice") && $this->getConfig->get("economy")) {
```

Comment:
All the code between the if check above and the else below can be massively simplified.